pennylane/math/matrix_manipulation.py

```diff
@@ -140,6 +145,37 @@ def expand_matrix(base_matrix, wires, wire_order=None, sparse_format="csr"):
     return mat
 
 
+def reduce_matrices(
+    mats_and_wires_gen: Generator[Tuple[np.ndarray, Wires], None, None], reduce_func: callable
+) -> Tuple[np.ndarray, Wires]:
+    """Apply the given ``reduce_func`` cumulatively to the items of the ``mats_and_wires_gen``
+    generator, from left to right, so as to reduce the sequence to a tuple containing a single
+    matrix and the wires it acts on.
+
+    Args:
+        mats_and_wires_gen (Generator): generator of tuples containing the matrix and the wires of
+            each operator
+        reduce_func (callable): function used to reduce the sequence of operators
+
+    Returns:
+        Tuple[tensor, Wires]: a tuple containing the reduced matrix and the wires it acts on
+    """
+
+    def expand_and_reduce(op1_tuple: Tuple[np.ndarray, Wires], op2_tuple: Tuple[np.ndarray, Wires]):
+        mat1, wires1 = op1_tuple
+        mat2, wires2 = op2_tuple
+        expanded_wires = wires1 + wires2
+        if wires1 != expanded_wires:
```

Comment:
See https://github.com/PennyLaneAI/pennylane/blob/acec4a11707ced0978c19534bcfd42160a9c0662/pennylane/math/matrix_manipulation.py#L99

`expand_matrix` performs a check to see if `wires1 == expanded_wires`.  If so, it returns early without performing and operations.

Reply:
You are right! Removed checks.